runtime/src/iree/tooling/vm_util.c

```diff
@@ -242,29 +260,35 @@ Status ParseToVariantList(iree_hal_allocator_t* device_allocator,
                                   (int)input_view.size, input_view.data);
         }
       }
-      IREE_RETURN_IF_ERROR(iree_vm_list_push_value(variant_list.get(), &val));
+      IREE_RETURN_AND_END_ZONE_IF_ERROR(
+          z0, iree_vm_list_push_value(variant_list, &val));
     }
   }
-  *out_list = variant_list.release();
-  return OkStatus();
+  *out_list = variant_list;
+  return iree_ok_status();
 }
 
 // Prints a buffer view with contents without a trailing newline.
-static iree_status_t PrintBufferView(iree_hal_buffer_view_t* buffer_view,
-                                     iree_host_size_t max_element_count,
-                                     iree_string_builder_t* builder) {
-  std::string result_str(4096, '\0');
+static iree_status_t iree_print_buffer_view(iree_hal_buffer_view_t* buffer_view,
```

Comment:
iree_tooling_append_buffer_view_string shows how to do this without looping - we should move that into iree/hal/buffer_view_util (there's already the format and fprint in there, so having the string builder variant would be good) - could be `iree_hal_buffer_view_append` or something

Reply:
done